designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+
+### RawParse
+
+This... mostly seems to do what I think it needs to do? It should probably have a big fat comment noting that we're piggybacking off profile parsing (keeping the net5.0+ dependent logic in `ParseFolder`). Maybe renaming variables to `profileOrPlatform` for clarity.
+
+## class FrameworkConstants
+
+Source: src/NuGet.Core/NuGet.Frameworks/FrameworkConstants.cs
+
+### (new!) FrameworkPlatforms
+
+A new `internal static HashSet<string> FrameworkPlatforms` field should be added and populated with all valid platforms (`android`, `ios`, `macos`, `tvos`, `watchos`, `windows`). As of this writing, it's **unclear** what to do about Tizen and Unity, and whether they'll be regular Platforms or a TFI or whatever.
+
+## class CompatibilityProvider
+
+Source: src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs
+
+### IsCompatibleWithTargetCore
+
+This method needs to be modified such that if there's a Platform present in the `candidate`, the platform name has to match. Additionally, an `IsVersionCompatible(target.PlatformVersion, candidate.PlatformVersion)` should be made if both have a `PlatformVersion` included. This will need to check for null `PlatformVersion`s, as well.
+
+## class NuGetFrameworkFullComparer
+
+Source: src/NuGet.Core/NuGet.Frameworks/comparers/NuGetFrameworkFullComparer.cs
+
+### Equals
+
+An extra clause should be added here that checks that `Platform` and `PlatformVersion` are compared against.
+
+Additionally, there's maybe some leftover work, as per [this comment in PR #3265](https://github.com/NuGet/NuGet.Client/pull/3265/files#r388004831) as far as what to do with `Unsupported`. This might be getting worked on separately?
+
+## class CompatibilityTests
+
+Source: test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs
+
+`[Theory] Compatibility_FrameworksAreCompatible` data needs to be modified such that Platforms are supported, and all expected platforms are included in the tests.
+
+## class NuGetFrameworkParseTests
+
+Source: test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkParseTests.cs
+
+`[Theory] NuGetFramework_Basic` needs additional test cases to make sure Platform and PlatformVersion are being parsed correctly from the "long" tfm format.
+
+## class NuGetFrameworkTests
+
+Source: test/NuGet.Core.Tests/NuGet.Frameworks.Test/NuGetFrameworkTests.cs
+
+An additional test should be added to make sure things like `net50-android10` parse to `net5.0-android10.0`.
+
+## Remaining work/questions
+
+1. Ensure Pack target passes TargetPlatformVersion to .nuspec generator. (Use `0.0` as a sentinel)
+1. Fail parsing/creation of `NuGetFramework` when an unknown OS flavor is being used -- slam an Unsupported tfi into it??

```

Comment:
I'm not sure. I'd like others opinions. In a future .NET, if they want to support new platforms, should they be dependant on NuGet updating our platform whitelist, before they can even progress on other parts of the build system?  I'm happy for NuGetFramework's `IsSupported` to return false if it's not a whitelisted platform, but I just don't know how I feel about returning the unsupported TFI.